otter/integration/tests/test_convergence.py

```diff
@@ -0,0 +1,212 @@
+                        "imageRef": image_ref,
+                    }
+                }
+            },
+            "groupConfiguration": {
+                "name": "converger-test-configuration",
+                "cooldown": 0,
+                "minEntities": N_SERVERS,
+            },
+            "scalingPolicies": [],
+        }
+
+        self.scaling_group = ScalingGroup(
+            group_config=scaling_group_body,
+            pool=self.pool
+        )
+
+        self.scaling_policy = ScalingPolicy(
+            scale_by=1,
+            scaling_group=self.scaling_group
+        )
+
+        return (
+            self.identity.authenticate_user(rcs)
+            .addCallback(find_end_points)
+            .addCallback(self.scaling_group.start, self)
+            .addCallback(
+                self.scaling_group.wait_for_N_servers,
+                N_SERVERS, timeout=1800
+            ).addCallback(self.scaling_group.get_scaling_group_state)
+            .addCallback(self._choose_half_the_servers)
+            .addCallback(self._delete_those_servers, rcs)
+            .addCallback(self.scaling_policy.start, self)
+            .addCallback(self.scaling_policy.execute)
+            .addCallback(self._wait_for_autoscale_to_catch_up, rcs)
+        )
+    test_reaction_to_oob_server_deletion.timeout = 600
+
+    def _choose_half_the_servers(self, t):
+        """Select the first half of the servers returned by the
+        ``get_scaling_group_state`` function.  Record the number of servers
+        received in ``n_servers`` attribute, and the number killed (which
+        should be roughly half) in ``n_killed``.
+        """
+
+        if t[0] != 200:
+            raise Exception("Got 404; where'd the scaling group go?")
+        ids = map(lambda obj: obj['id'], t[1]['group']['active'])
+        self.n_servers = len(ids)
+        self.n_killed = self.n_servers / 2
+        return ids[:self.n_killed]
+
+    def _delete_those_servers(self, ids, rcs):
+        """Delete each of the servers selected."""
+
+        def delete_server_by_id(i):
+            return (
+                treq.delete(
+                    "{}/servers/{}".format(str(rcs.endpoints["nova"]), i),
+                    headers=headers(str(rcs.token)),
+                    pool=self.pool
+                ).addCallback(check_success, [204])
+                .addCallback(lambda _: rcs)
+            )
+
+        deferreds = map(delete_server_by_id, ids)
+        # If no error occurs while deleting, all the results will be the
+        # same.  So just return the 1st, which is just our rcs value.
+        return gatherResults(deferreds).addCallback(lambda rslts: rslts[0])
+
+    def _wait_for_autoscale_to_catch_up(self, _, rcs, timeout=60, period=1):
+        """Wait for the converger to recognize the reality of this tenant's
+        situation and reflect it in the scaling group state accordingly.
+        """
+
+        def check(state):
+            if state[0] != 200:
```

Comment:
Nitpick: Could you take arguments like `def check((code, response))` and check for code or
```python
code, response = state
if code != 200:
```
This makes it easier to understand.